lib/cequel/record/associations.rb

```diff
@@ -108,14 +108,16 @@ def belongs_to(name, options = {})
                  "belongs_to association must be declared before declaring " \
                  "key(s)"
           end
-          
+
           key_options = options.extract!(:partition)
+          foreign_key_parts = [options.fetch(:foreign_key, [])].flatten
 
           self.parent_association =
             BelongsToAssociation.new(self, name.to_sym, options)
 
-          parent_association.association_key_columns.each do |column|
-            key :"#{name}_#{column.name}", column.type, key_options
+          parent_association.association_key_columns.each_with_index do |column, i|
+            foreign_key = foreign_key_parts[i].nil? ? "#{name}_#{column.name}" : foreign_key_parts[i]
+            key foreign_key.to_sym, column.type, key_options
```

Comment:
I updated the example in markdown above to include an attribute called account_id (on Parent) which is outside of our domain. This illustrates the use of the array foreign_key.